### PR TITLE
Update to latest http-client

### PR DESCRIPTION
**Pull request by dmalikov**
Not sure how to update to latest `liblastfm` and `http-client-0.5` having all the necessary functionality :(

### CODE REVIEW

**Comment by dmalikov** [comment]
It's also necessary to check "error" field in the response body - "11" / "16" values could be retried 

https://www.last.fm/api/show/track.scrobble

**Comment by supki** [comment]
Are you blocked on something or just figured it's not worth it
?

**Comment by dmalikov** [comment]
Yep, everything just works fine even without retrying. Frankly speaking, it's hard for me to decide if 11/16 should be retried or not.

**Comment by supki** [comment]
Well, the docs explicitly say to retry in case of 11 or 16, so I'd say it's reasonable. it's great that it works for you anyway, though!

File: src/Control/Scrobbler/Lastfm.hs [hunk 1]
```diff
@@ -57,16 +57,13 @@ scrobble Credentials { secret, apiKey, sessionKey } =
...
-        -- Otherwise if we have response
-        | Just w <- lookup "Response" hs ->
-          -- Check if it's still some server error, we hold the judgement on
-          if | server (fromStrict w) -> ([], toList tss)
-          -- Otherwise it was a client error and we drop the tracks
-          -- (something really bad happened like broken API wrapper or whatever)
-             | otherwise -> ([], [])
+        | (statusCode $ responseStatus response) >= 500 -> ([], toList tss)
+        -- Check it it's still some server error, we hold the judgement on
+        | (server chunk) -> ([], toList tss)
```
Comment (supki) [inline src/Control/Scrobbler/Lastfm.hs]: redundant parens

File: src/Control/Scrobbler/Lastfm.hs [hunk 2]
```diff
@@ -57,16 +57,13 @@ scrobble Credentials { secret, apiKey, sessionKey } =
...
         -- Status code >= 500 means server error, we hold our judgement on
-        | c >= 500 -> ([], toList tss)
-        -- Otherwise if we have response
-        | Just w <- lookup "Response" hs ->
-          -- Check if it's still some server error, we hold the judgement on
-          if | server (fromStrict w) -> ([], toList tss)
-          -- Otherwise it was a client error and we drop the tracks
-          -- (something really bad happened like broken API wrapper or whatever)
-             | otherwise -> ([], [])
+        | (statusCode $ responseStatus response) >= 500 -> ([], toList tss)
```
Comment (supki) [inline src/Control/Scrobbler/Lastfm.hs]: redundant `$`

File: .travis.yml [hunk 3]
```diff
@@ -1,15 +1,33 @@
-language: haskell
+# https://docs.haskellstack.org/en/stable/travis_ci/
```
Comment (supki) [inline .travis.yml]: Why do we test stuff with stack now?

File: src/Control/Scrobbler/Lastfm.hs [hunk 4]
```diff
@@ -78,7 +75,7 @@ scrobble Credentials { secret, apiKey, sessionKey } =
           , toList tss^..ifolded.indices (`elem` is)
           )
 
-  server    = maybe False (`elem` [11, 16]) . preview (key "error" . _Number)
+  server = maybe False (`elem` [11, 16]) . preview (key "error" . _Number)
```
Comment (supki) [inline src/Control/Scrobbler/Lastfm.hs]: Maybe name these magic constants? Apparently, it was a mistake not to name them earlier.

**Review by supki — COMMENTED** [review]
LGTM, apart from the sudden change to stack.

File: .travis.yml (hunk 3 above)
Comment (dmalikov) [inline .travis.yml]: At the moment, it's the only way I'm able to build this project at my laptop. `nix` infrastructure takes too much effort to maintain, especially with non-linux OS.
At the same time, even if `nix` works in Travis, it cannot build this particular project in a given time quota.

Anyway, let's give `stack` a chance.

File: src/Control/Scrobbler/Lastfm.hs (hunk 4 above)
Comment (dmalikov) [inline src/Control/Scrobbler/Lastfm.hs]: Sure, why not.

File: .travis.yml (hunk 3 above)
Comment (supki) [inline .travis.yml]: Oh, OK. Certainly, being able to build the project is pretty important! Sorry to hear you got yourself into that unfortunate non-linux situation.

**Pull request merged**